auth0/src/main/java/com/auth0/android/request/RetryInterceptor.kt

```diff
@@ -0,0 +1,46 @@
+package com.auth0.android.request
+
+import com.auth0.android.dpop.DPoPProvider
+import okhttp3.Interceptor
+import okhttp3.Response
+
+/**
+ * Interceptor that retries requests.
+ */
+internal class RetryInterceptor : Interceptor {
+    override fun intercept(chain: Interceptor.Chain): Response {
+        val request = chain.request()
+        val retryCountHeader = request.header(RETRY_COUNT_HEADER)
+        val currentRetryCount = retryCountHeader?.toIntOrNull() ?: 0
+        val response = chain.proceed(request)
+
+        //Handling DPoP Nonce retry
+        if (DPoPProvider.isNonceRequiredError(response) && currentRetryCount < DPoPProvider.MAX_RETRY_COUNT) {
+            DPoPProvider.storeNonce(response)
```

Comment:
The nonce can also be sent ahead of time in a 200 OK response (https://www.rfc-editor.org/rfc/rfc9449.html#name-providing-a-new-nonce-value), for example, in the `/userinfo` response.